include/versions.hh

```diff
@@ -45,12 +47,33 @@ bool isDate( const std::string & version );
 /** @return `true` iff @a version can be interpreted as _semantic version_. */
 bool isCoercibleToSemver( const std::string & version );
 
+/**
+ * @brief Determine if @a version is a valid _semantic version range_ string.
+ *
+ * This is far from a complete check, but it should be sufficient for our usage.
```

Comment:
What do we need to be correct for our usage? Would there be any options for checking this without implementing it ourselves?

Reply:
We could tweak `node-semver` to expose additional subcommands for parts of its parsers.

Similar to what we've done with `parser-util` and `flox`.

Reply:
If you're going to leave as is, can you explain why it's sufficient for our usage? I.e. if we need to change it, what kinds of things need to work? And what kinds of things can be incomplete?

Reply:
It's sufficient for a first iteration.
It checks the 99% case to distinguish between a range and "static" version.

The main reason to detect this is because from the CLI we can't immediately tell whether `<NAME>@<VERSION-OR-SEMVER>` is an exact version match ( like a date ), or a real range.

This does a "best effort" detection which is fine for now in my opinion.

Reply:
Cool can you add a comment saying it's checking the 99% case which is _, and not the 1% case which is _? I'm just wanting to understand so that if we have to touch it without you we'll have some idea of what needs to work

Reply:
I expanded the inline comment

Reply:
Thanks!